docs/configuration.rst

```diff
@@ -60,34 +64,65 @@ An example of custom configuration that allows OpenTAXII to connect to productio
-    - ``unauthorized_status`` — TAXII status type for authorization error. "UNAUTHORIZED" by default. see `libtaxii.constants.ST_TYPES_11 <https://libtaxii.readthedocs.io/en/stable/api/constants.html#libtaxii.constants.ST_TYPES_11>`_ for the list of available values.
-    - ``persistence_api`` — configuration properties for Persistence API implementation.
     - ``auth_api`` — configuration properties for Authentication API implementation.
+
+      - ``class`` — the full import name of the class to use
+      - ``parameters`` — the parameters used to instantiate the class
+
+        - ``db_connection`` — the database connetion string
+        - ``create_tables`` — boolean, if true, create tables on startup
+        - ``secret`` — the secret with which the generated tokens are encoded
+        - ``token_ttl_secs`` — time that generated tokens are valid
+
+    - ``taxii1`` — taxii1-specific settings
+
+      - ``save_raw_inbox_message`` — enable/disable storing of raw TAXII Inbox messages via Persistence API's ``create_inbox_message`` method. This is useful for bookkeeping but significantly increases storage requirements.
+      - ``xml_parser_supports_huge_tree`` — enable/disable security restrictions in `lxml <http://lxml.de/>`_ library to allow support for very deep trees and very long text content. If this is disabled, OpenTAXII will not be able to parse TAXII messages with content blocks larger than roughly 10MB.
+      - ``count_blocks_in_poll_responses`` — enable/disable total count in TAXII Poll responses. It is disabled by default since ``count`` operation might be `very slow <https://wiki.postgresql.org/wiki/Slow_Counting>`_ in some SQL DBs.
+      - ``unauthorized_status`` — TAXII status type for authorization error. "UNAUTHORIZED" by default. see `libtaxii.constants.ST_TYPES_11 <https://libtaxii.readthedocs.io/en/stable/api/constants.html#libtaxii.constants.ST_TYPES_11>`_ for the list of available values.
+      - ``hooks`` - custom python module with signal subscriptions to import. See :ref:`documentation on custom signals<custom-signals>` and :github-file:`an example <examples/hooks.py>`.
+      - ``persistence_api`` — configuration properties for Persistence API implementation.
+
+        - ``class`` — the full import name of the class to use
+        - ``parameters`` — the parameters used to instantiate the class
+
+          - ``db_connection`` — the database connetion string
+          - ``create_tables`` — boolean, if true, create tables on startup
+
+    - ``taxii2`` — taxii2-specific settings
+
+      - ``persistence_api`` — configuration properties for Persistence API implementation.
+
+        - ``class`` — the full import name of the class to use
+        - ``parameters`` — the parameters used to instantiate the class
+
+          - ``db_connection`` — the database connetion string
+          - ``create_tables`` — boolean, if true, create tables on startup
```

Comment:
I seperated them on purpose, because the taxii1 persistence api methods don't make sense for taxii2 and vice versa. So while the keys are duplicated, the values should be different (taxii1 should be a subclass of `OpenTAXIIPersistenceAPI` and taxii2 should be a subclass of `OpenTAXII2PersistenceAPI`).